src/ui/actions/ActionResult.cpp

```diff
@@ -0,0 +1,41 @@
+//
+// Created by Anton O. on 12/18/21.
+//
+
+#include "ActionResult.h"
+#include "utilities/ModelInquiryResultUtils.h"
+
+ActionResult::ActionResult(const Core::ModelInquiryResult &result) :
+              model_result{result}, type_id{kResult} {
+}
+
+ActionResult::ActionResult(const Core::TaskEntity &te) :
+              entity{te}, type_id{kEntity} {
+}
+
+ActionResult::ActionResult(const std::vector<Core::TaskEntity> &vec) :
+              tasks{vec}, type_id{kVector} {
+}
+
+ActionResult::~ActionResult() {
+    switch (type_id) {
+        case kResult:
+            (&model_result)->Core::ModelInquiryResult::~ModelInquiryResult();
```

Comment:
Maybe not the best solution.

Reply:
Already comment in other review: Never call destructors directly!!!!!!!!

Reply:
Fixed - refused using union altogether, employed std::optional instead